doc/developer/platform/ux.md

```diff
@@ -1,20 +1,945 @@
+
+    `PUT /api/region/{cloud-provider}/{region}`
+
+In the future, we will add billing operations to this API.
+
+As with the admin API, we don't plan to publicize or document the global API,
+but we don't plan to hide it, either.
+
+### Region API
+
+The user-facing API for Materialize Platform is the region API. This is the
+only API that we will actively advertise to users. The intent is that users
+think of the region API as **the** Materialize Platform API.
+
+**The region API is a SQL API.** It is not a REST API. While the API will
+be accessible via HTTPS, HTTPS is merely a container for a SQL payload.
+This is likely to be a contentious topic. Please see [Why SQL?](#why-sql)
+for justification.
+
+This implies that the web UI will communicate with the region API by using
+the SQL-over-HTTPS API.
+
+#### Endpoint
+
+The endpoint for the region API is a hostname of the following form:
+
+> CUSTOMER-ID.REGION.CLOUD-PROVIDER.materialize.cloud
+
+For example:
+
+> x32sk13nf.us-east-1.aws.materialize.cloud
+
+This endpoint requires TLS encryption. The server will present a TLS certificate
+issued by a trusted root CA.
+
+#### PostgreSQL server
+
+A PostgreSQL protocol server will be accessible at port 6875. The server
+processes SQL as described in [PostgreSQL Frontend/Backend Protocol][pgwire].
+
+The PostgreSQL server will support only username/password authentication. The
+username must be the email address of a [user](#user) in the account. The
+password must be a valid [app-specific password](#app-specific-password) for the
+user. This is notably *not* the password that the user would supply to log into
+the web UI, as that password is configured to additionally require two-factor
+authentication and CAPTCHA completion.
+
+The existing Materialize Cloud support for client TLS authentication will be
+removed, with no plans for restoration.
+
+We do not currently expect to need to make any changes to the existing
+PostgreSQL protocol server in `materialized`.
+
+#### HTTPS
+
+An HTTPS server will be accessible at port 443.
+
+To authenticate with the HTTPS server, users supply an authentication token
+obtained from the [admin API](#admin-api).
+
+The HTTPS server exposes two APIs:
+
+  * `POST /query` executes a batch of SQL statements.
+
+    The request body specifies the SQL string to execute. The request may contain
+    multiple SQL statements separated by semicolons. The statements are submitted
+    for processing as if they were supplied via the [Simple Query][simple-query]
+    flow of the PostgreSQL protocol (i.e., "the statements are executed as a
+    single transaction unless explicit transaction control commands are included
+    to force a different behavior").
+
+    The response is a JSON array containing the results of each query in the
+    request.
+
+    TODO: fully describe the output format.
+
+  * `ws://<ENDPOINT>/query?query=QUERY` streams the result of a single query
```

Comment:
Might be reasonable to accept the queries as messages from the ws client instead of in the URL itself, for full interactivity.